projects/sbb-esta/angular-business/notification/src/notification/notification.component.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  EventEmitter,
+  HostBinding,
+  Input,
+  Output,
+  TemplateRef,
+  ViewChild,
+  ViewEncapsulation
+} from '@angular/core';
+import { IconDirective } from '@sbb-esta/angular-core/icon-directive';
+
+export enum NotificationType {
+  SUCCESS = 'success',
+  ERROR = 'error',
+  INFO = 'info',
+  WARN = 'warn'
+}
+
+export enum NotificationToastPosition {
+  TOPLEFT = 'top-left',
+  TOPRIGHT = 'top-right',
+  BOTTOMLEFT = 'bottom-left',
+  BOTTOMRIGHT = 'bottom-right'
+}
+
+export interface JumpMark {
+  /** Title of an element in jump marks. */
+  title: string;
+  /** Identifier of an element in jump marks. */
+  elementId?: string;
+  callback?: (event$: any, jumpMark: JumpMark) => void;
+}
+
+@Component({
+  selector: 'sbb-notification',
+  templateUrl: './notification.component.html',
+  styleUrls: ['./notification.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class NotificationComponent {
+  /** @docs-private */
+  @HostBinding('class.sbb-notification')
+  baseCssClass = true;
+
+  /** @docs-private */
+  @HostBinding('class.sbb-notification-success')
+  get typeSuccess(): boolean {
+    return this.type === NotificationType.SUCCESS;
+  }
+
+  /** @docs-private */
+  @HostBinding('class.sbb-notification-info')
+  get typeInfo(): boolean {
+    return this.type === NotificationType.INFO;
+  }
+
+  /** @docs-private */
+  @HostBinding('class.sbb-notification-error')
+  get typeError(): boolean {
+    return this.type === NotificationType.ERROR;
+  }
+
+  /** @docs-private */
+  @HostBinding('class.sbb-notification-warn')
+  get typeWarn(): boolean {
+    return this.type === NotificationType.WARN;
+  }
+
+  /** @docs-private */
+  @HostBinding('class.sbb-notification-toast-top-left')
+  get positionTopLeft(): boolean {
+    return this.toastPosition === NotificationToastPosition.TOPLEFT;
+  }
+
+  /** @docs-private */
+  @HostBinding('class.sbb-notification-toast-top-right')
+  get positionTopRight(): boolean {
+    return this.toastPosition === NotificationToastPosition.TOPRIGHT;
+  }
+
+  /** @docs-private */
+  @HostBinding('class.sbb-notification-toast-bottom-left')
+  get positionBottomLeft(): boolean {
+    return this.toastPosition === NotificationToastPosition.BOTTOMLEFT;
+  }
+
+  /** @docs-private */
+  @HostBinding('class.sbb-notification-toast-bottom-right')
+  get positionBottomRight(): boolean {
+    return this.toastPosition === NotificationToastPosition.BOTTOMRIGHT;
+  }
+
+  /** Type of notification. */
+  @Input()
+  type: 'success' | 'info' | 'error' | 'warn' = NotificationType.SUCCESS;
+
+  /** Type of notification. */
+  @Input()
+  toastPosition: 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
+
+  /** Type of notification. */
+  @Input()
+  readonly = false;
```

Comment:
use the `coerceBooleanProperty` pattern here
```suggestion
  @Input()
  set readonly(value: boolean) {
    this._readonly = coerceBooleanProperty(value);
    this._changeDetectorRef.markForCheck();
  }
  get readonly() {
    return this._readonly;
  }
  private _readonly = false;
```

Reply:
Fixed